client/src/request.rs

```diff
@@ -267,4 +306,115 @@ impl SuccinctClient {
             Err(Error::msg("Failed to submit request to Succinct X API."))
         }
     }
+
+    /// Submit a request to the Succinct X API.
+    /// If in local prove mode, generates a local proof and returns the request_id after completion.
+    pub async fn submit_request(
+        &self,
+        chain_id: u32,
+        to: Address,
+        calldata: Bytes,
+        function_id: B256,
+        input: Bytes,
+    ) -> Result<String> {
+        if self.local_prove_mode {
+            return self.submit_local_request(
+                chain_id,
+                to,
+                calldata.clone(),
+                function_id,
+                input.clone(),
+            );
+        }
+
+        self.submit_platform_request(chain_id, to, calldata, function_id, input)
+            .await
+    }
+
+    /// If in local relay mode, ethereum_rpc_url and wallet must be provided. If you wish to submit
+    /// to your own gateway (ex. on a chain that doesn't have a canonical gateway), gateway_address
+    /// must be provided.
+    // TODO: Add support for hosted proving + local relaying.
+    pub async fn relay_proof(
+        &self,
+        request_id: String,
+        ethereum_rpc_url: Option<&str>,
+        wallet: Option<LocalWallet>,
+        gateway_address: Option<&str>,
+    ) -> Result<()> {
+        // If local mode, submit proof from local directory at proofs/output_{request_id}.json
+        if self.local_relay_mode {
```

Comment:
Why would you ever call this function if `self.local_relay_mode` is false, I don't see the point of just wrapping the entire function in the flag?

Reply:
Yeah this should be local prove mode